test/EndToEndTests/Common/Microsoft.OData.Client.E2E.TestCommon/EndToEndTestBaseOfT.cs

```diff
@@ -0,0 +1,58 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="EndToEndTestBaseOfT.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.Extensions.DependencyInjection;
+using Xunit;
+
+namespace Microsoft.OData.Client.E2E.TestCommon
+{
+    public class EndToEndTestBase<T> : IClassFixture<TestWebApplicationFactory<T>> where T : class
+    {
+        private HttpClient _client;
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebODataTestBase{TStartup}"/> class.
+        /// </summary>
+        /// <param name="factory">The factory used to initialize the web service client.</param>
+        protected EndToEndTestBase(TestWebApplicationFactory<T> factory)
+        {
+            Factory = factory ?? throw new ArgumentNullException(nameof(factory));
+            _httpClientFactory = Factory.Services.GetRequiredService<IHttpClientFactory>();
+        }
+
+        /// <summary>
+        /// An HttpClient to use with the server.
+        /// </summary>
+        public virtual HttpClient Client
+        {
+            get
+            {
+                if (_client == null)
+                {
+                    _client = Factory.CreateClient();
+                }
+                return _client;
+            }
+        }
+
+        /// <summary>
+        /// Gets the factory.
+        /// </summary>
+        public TestWebApplicationFactory<T> Factory { get; }
+
+        /// <summary>
+        /// Gets the HttpClientFactory.
+        /// </summary>
+        public IHttpClientFactory HttpClientFactory => _httpClientFactory;
+
+        /// <summary>
+        /// Gets or sets the delegate for creating a DataServiceContext for the service.
+        /// </summary>
+        public Func<Uri, DataServiceContext> CreateDataServiceContext { get; set; }

```

Comment:
Since this delegate is not used anywhere in this class, I don't think it serves any purpose. I think this property should be removed.